tools/dwca.rb

```diff
@@ -93,11 +93,23 @@
   opt :with_controlled_values, "Only include observations with annotations with this value (must be combined with `with_controlled_terms`)", type: :strings
   opt :include_humans, "Include observations of humans", type: :boolean, default: false
   opt :processes, "Number of processes to use with the parallel gem", type: :integer
+  opt :log_task_name, "Log with the specified task name", type: :string
 end
 
 if opts.debug
   opts[:logger] = Logger.new(STDOUT, level: Logger::DEBUG)
 else
   opts[:logger] = Logger.new(STDOUT, level: Logger::INFO)
 end
-DarwinCore::Archive.generate(opts)
+
+if opts.log_task_name
+  task_logger = TaskLogger.new( opts.log_task_name, nil, "export", "rails" )
+end
+
+begin
+  task_logger&.start
+  DarwinCore::Archive.generate( opts )
+  task_logger&.end
+rescue => e
+  task_logger&.error( e.to_s )
```

Comment:
Might consider catching a few lines of the error `backtrace` as well if we want to use this to pinpoint where errors are occurring. If so, make sure that if you use newlines to separate backtrace lines, those are escaped and don't interfere with keeping each task logger log entry on a single line in the log file